src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/ZeroFreeCoTaskMemUnicodeTests.cs

```diff
@@ -0,0 +1,24 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Runtime.InteropServices.Tests
+{
+    public class ZeroFreeCoTaskMemUnicodeTests
+    {
+        [Fact]
+        public void ZeroFreeCoTaskMemUnicode_ValidPointer_Success()
+        {
+            IntPtr ptr = Marshal.StringToCoTaskMemUni("hello");
+            Marshal.ZeroFreeCoTaskMemUnicode(ptr);
```

Comment:
From https://github.com/dotnet/corefx/pull/31057

@luqunl

> this is incorrect. memory allocated by Marshal.StringToHGlobalUni should be released by Marshal.FreeHGlobal

Reply:
@luqunl 
> similar to others. memory allocated by SecureStringToGlobalAllocUnicode should be released by Marshal.ZeroFreeGlobalAllocUnicode

Reply:
Looking at the implementation:
```cs
public static void ZeroFreeGlobalAllocUnicode(IntPtr s)
{
    RuntimeImports.RhZeroMemory(s, (UIntPtr)(Win32Native.lstrlenW(s) * 2));
    FreeHGlobal(s);
}
```

I don't see an issue with this usage here? /cc @AaronRobinsonMSFT

Reply:
@hughbe I agree with you that technically this is perfectly acceptable. However, I think @luqunl's point is that based on the documentation. Consider `Marshal.ZeroFreeCoTaskMemUnicode()`, the documentation states that `Marshal.SecureStringToCoTaskMemUnicode()` should be used, where as `Marshal.StringToCoTaskMemUni()` states `Marshal.FreeCoTaskMem()`.

The general guidance should be that whatever the documentation states is what should be followed, unless there is some coreclr deviation or internal knowledge - which we should then document.

Reply:
Then having the API `ZeroFreeCoTaskMemUTF8` is even more inconsistent as we don't have  an `SecureString` overloads!